### PR TITLE
Uf 6580 integrate towards samba share

### PR DESCRIPTION
**Pull request by mha109**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -10,14 +10,16 @@
...
-    <version>1.3</version>
+    <version>2.0</version>
     <name>api-service-snailmail-sender</name>
     <properties>
         <!-- Pipeline properties -->
         <docker.image.name>ms-snailmail-sender</docker.image.name>
         <published.container.port.test>9077</published.container.port.test>
         <published.container.port.production>9077</published.container.port.production>
         <!-- Generated sources properties -->
+        <openapi-generator-maven-plugin>7.1.0</openapi-generator-maven-plugin>
```
Comment (foobar-1337) [inline pom.xml:21]: Remove

File: src/main/java/se/sundsvall/snailmail/api/SnailMailResource.java [hunk 2]
```diff
@@ -33,14 +33,13 @@ public SnailMailResource(final SnailMailService snailMailService) {
 		this.snailMailService = snailMailService;
 	}
 
-	@PostMapping(consumes = APPLICATION_JSON_VALUE)
+	@PostMapping(path = "/snailmail", consumes = APPLICATION_JSON_VALUE)
 	@Operation(summary = "Create snailmail")
```
Comment (cheezi0747) [inline src/main/java/se/sundsvall/snailmail/api/SnailMailResource.java:37]: "Create snailmail"  --> "Prepare a snailmail for batch" or something perhaps?

File: src/main/java/se/sundsvall/snailmail/integration/citizen/CitizenClient.java [hunk 3]
```diff
@@ -19,6 +19,11 @@
 @CircuitBreaker(name = CitizenIntegration.INTEGRATION_NAME)
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/snailmail/integration/citizen/CitizenClient.java:19]: Remove

File: src/main/java/se/sundsvall/snailmail/api/model/SendSnailMailRequest.java [hunk 4]
```diff
@@ -18,12 +22,21 @@
...
 	@NotBlank
+	@ValidFolderName
 	@Schema(description = "Department and unit that should be billed", example = "SBK(Gatuavdelningen, Trafiksektionen)")
 	private String department;
 
 	@Schema(description = "If the letter to send deviates from the standard", example = "A3 Ritning")
 	private String deviation;
 
+	@NotBlank
+	@Schema(description = "Batch id to be used for creating a csv-file")
```
Comment (cheezi0747) [inline src/main/java/se/sundsvall/snailmail/api/model/SendSnailMailRequest.java:33]: example

File: src/main/java/se/sundsvall/snailmail/api/model/SendSnailMailRequest.java [hunk 5]
```diff
@@ -40,8 +53,8 @@ public static class Attachment {
 		@NotBlank
 		private String name;
 
-		@Schema(description = "The attachment content type", example = "text/plain")
-		@NotBlank
+		@OneOf("application/pdf")
+		@Schema(description = "The attachment content type", allowableValues = {"application/pdf"})
```
Comment (cheezi0747) [inline src/main/java/se/sundsvall/snailmail/api/model/SendSnailMailRequest.java:57]: example

**Review by cheezi0747 — APPROVED** [review]

**Review by cheezi0747 — APPROVED** [review]

**Pull request merged**